pkg/types/validation/installconfig.go

```diff
@@ -307,6 +307,9 @@ func validateControlPlane(platform *types.Platform, pool *types.MachinePool, fld
 	if pool.Replicas != nil && *pool.Replicas == 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "number of control plane replicas must be positive"))
 	}
+	if pool.Platform.Azure != nil && pool.Name == masterPoolName && pool.Platform.Azure.OSDisk.DiskType == "Standard_LRS" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("diskType"), pool.Platform.Azure.OSDisk.DiskType, "DiskType not compatible with master machines."))
+	}
```

Comment:
i think this is redundant, as you already have https://github.com/openshift/installer/pull/3520/files#diff-e8044151bf8d8c8e84c04c49911da440R64-R66